nestkernel/conn_builder.cpp

```diff
@@ -335,10 +329,8 @@ nest::ConnBuilder::create_param_dict_( index snode_id,
         ( *dd ) = synapse_parameter.second->value_double( target_thread, rng, snode_id, &target );
       }
     }
-    param_dict = param_dicts_[ indx ][ target_thread ];
+    dummy_param_dicts_[ target_thread ] = param_dicts_[ indx ][ target_thread ];
```

Comment:
If I understand the entire context right, then `dummy_param_dicts_` should be an empty dictionary if for a given synapse spec there are no synapse parameters (except weight and delay, which are handled explicitly). With the code as it is now, I think that if we have collocated synapses with specs A and B, where A contains additional parameters and B not, then this code will set `dummy_params_dicts` to a non-empty dictionary when processing A, but not do anything when processing B, so B will be created with the parameter dict from A still in `dummy_param_dicts_`, which is wrong. Or am I overlooking something? Do we have a test for that case?